// @json-render/shadcn/dist/catalog.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/catalog.ts
var catalog_exports = {};
__export(catalog_exports, {
  shadcnComponentDefinitions: () => shadcnComponentDefinitions
});
module.exports = __toCommonJS(catalog_exports);
var import_zod = require("zod");
var validationCheckSchema = import_zod.z.array(
  import_zod.z.object({
    type: import_zod.z.string(),
    message: import_zod.z.string(),
    args: import_zod.z.record(import_zod.z.string(), import_zod.z.unknown()).optional()
  })
).nullable();
var validateOnSchema = import_zod.z.enum(["change", "blur", "submit"]).nullable();
var shadcnComponentDefinitions = {
  // ==========================================================================
  // Layout Components
  // ==========================================================================
  Card: {
    props: import_zod.z.object({
      title: import_zod.z.string().nullable(),
      description: import_zod.z.string().nullable(),
      maxWidth: import_zod.z.enum(["sm", "md", "lg", "full"]).nullable(),
      centered: import_zod.z.boolean().nullable()
    }),
    slots: ["default"],
    description: "Container card for content sections. Use for forms/content boxes, NOT for page headers.",
    example: { title: "Overview", description: "Your account summary" }
  },
  Stack: {
    props: import_zod.z.object({
      direction: import_zod.z.enum(["horizontal", "vertical"]).nullable(),
      gap: import_zod.z.enum(["none", "sm", "md", "lg"]).nullable(),
      align: import_zod.z.enum(["start", "center", "end", "stretch"]).nullable(),
      justify: import_zod.z.enum(["start", "center", "end", "between", "around"]).nullable()
    }),
    slots: ["default"],
    description: "Flex container for layouts",
    example: { direction: "vertical", gap: "md" }
  },
  Grid: {
    props: import_zod.z.object({
      columns: import_zod.z.number().nullable(),
      gap: import_zod.z.enum(["sm", "md", "lg"]).nullable()
    }),
    slots: ["default"],
    description: "Grid layout (1-6 columns)",
    example: { columns: 3, gap: "md" }
  },
  Separator: {
    props: import_zod.z.object({
      orientation: import_zod.z.enum(["horizontal", "vertical"]).nullable()
    }),
    description: "Visual separator line"
  },
  Tabs: {
    props: import_zod.z.object({
      tabs: import_zod.z.array(
        import_zod.z.object({
          label: import_zod.z.string(),
          value: import_zod.z.string()
        })
      ),
      defaultValue: import_zod.z.string().nullable(),
      value: import_zod.z.string().nullable()
    }),
    slots: ["default"],
    events: ["change"],
    description: "Tab navigation. Use { $bindState } on value for active tab binding."
  },
  Accordion: {
    props: import_zod.z.object({
      items: import_zod.z.array(
        import_zod.z.object({
          title: import_zod.z.string(),
          content: import_zod.z.string()
        })
      ),
      type: import_zod.z.enum(["single", "multiple"]).nullable()
    }),
    description: "Collapsible sections. Items as [{title, content}]. Type 'single' (default) or 'multiple'."
  },
  Collapsible: {
    props: import_zod.z.object({
      title: import_zod.z.string(),
      defaultOpen: import_zod.z.boolean().nullable()
    }),
    slots: ["default"],
    description: "Collapsible section with trigger. Children render inside."
  },
  Dialog: {
    props: import_zod.z.object({
      title: import_zod.z.string(),
      description: import_zod.z.string().nullable(),
      openPath: import_zod.z.string()
    }),
    slots: ["default"],
    description: "Modal dialog. Set openPath to a boolean state path. Use setState to toggle."
  },
  Drawer: {
    props: import_zod.z.object({
      title: import_zod.z.string(),
      description: import_zod.z.string().nullable(),
      openPath: import_zod.z.string()
    }),
    slots: ["default"],
    description: "Bottom sheet drawer. Set openPath to a boolean state path. Use setState to toggle."
  },
  Carousel: {
    props: import_zod.z.object({
      items: import_zod.z.array(
        import_zod.z.object({
          title: import_zod.z.string().nullable(),
          description: import_zod.z.string().nullable()
        })
      )
    }),
    description: "Horizontally scrollable carousel of cards."
  },
  // ==========================================================================
  // Data Display Components
  // ==========================================================================
  Table: {
    props: import_zod.z.object({
      columns: import_zod.z.array(import_zod.z.string()),
      rows: import_zod.z.array(import_zod.z.array(import_zod.z.string())),
      caption: import_zod.z.string().nullable()
    }),
    description: 'Data table. columns: header labels. rows: 2D array of cell strings, e.g. [["Alice","admin"],["Bob","user"]].',
    example: {
      columns: ["Name", "Role"],
      rows: [
        ["Alice", "Admin"],
        ["Bob", "User"]
      ]
    }
  },
  Heading: {
    props: import_zod.z.object({
      text: import_zod.z.string(),
      level: import_zod.z.enum(["h1", "h2", "h3", "h4"]).nullable()
    }),
    description: "Heading text (h1-h4)",
    example: { text: "Welcome", level: "h1" }
  },
  Text: {
    props: import_zod.z.object({
      text: import_zod.z.string(),
      variant: import_zod.z.enum(["body", "caption", "muted", "lead", "code"]).nullable()
    }),
    description: "Paragraph text",
    example: { text: "Hello, world!" }
  },
  Image: {
    props: import_zod.z.object({
      src: import_zod.z.string().nullable(),
      alt: import_zod.z.string(),
      width: import_zod.z.number().nullable(),
      height: import_zod.z.number().nullable()
    }),
    description: "Image component. Renders an img tag when src is provided, otherwise a placeholder."
  },
  Avatar: {
    props: import_zod.z.object({
      src: import_zod.z.string().nullable(),
      name: import_zod.z.string(),
      size: import_zod.z.enum(["sm", "md", "lg"]).nullable()
    }),
    description: "User avatar with fallback initials",
    example: { name: "Jane Doe", size: "md" }
  },
  Badge: {
    props: import_zod.z.object({
      text: import_zod.z.string(),
      variant: import_zod.z.enum(["default", "secondary", "destructive", "outline"]).nullable()
    }),
    description: "Status badge",
    example: { text: "Active", variant: "default" }
  },
  Alert: {
    props: import_zod.z.object({
      title: import_zod.z.string(),
      message: import_zod.z.string().nullable(),
      type: import_zod.z.enum(["info", "success", "warning", "error"]).nullable()
    }),
    description: "Alert banner",
    example: {
      title: "Note",
      message: "Your changes have been saved.",
      type: "success"
    }
  },
  Progress: {
    props: import_zod.z.object({
      value: import_zod.z.number(),
      max: import_zod.z.number().nullable(),
      label: import_zod.z.string().nullable()
    }),
    description: "Progress bar (value 0-100)",
    example: { value: 65, max: 100, label: "Upload progress" }
  },
  Skeleton: {
    props: import_zod.z.object({
      width: import_zod.z.string().nullable(),
      height: import_zod.z.string().nullable(),
      rounded: import_zod.z.boolean().nullable()
    }),
    description: "Loading placeholder skeleton"
  },
  Spinner: {
    props: import_zod.z.object({
      size: import_zod.z.enum(["sm", "md", "lg"]).nullable(),
      label: import_zod.z.string().nullable()
    }),
    description: "Loading spinner indicator"
  },
  Tooltip: {
    props: import_zod.z.object({
      content: import_zod.z.string(),
      text: import_zod.z.string()
    }),
    description: "Hover tooltip. Shows content on hover over text."
  },
  Popover: {
    props: import_zod.z.object({
      trigger: import_zod.z.string(),
      content: import_zod.z.string()
    }),
    description: "Popover that appears on click of trigger."
  },
  // ==========================================================================
  // Form Input Components
  // ==========================================================================
  Input: {
    props: import_zod.z.object({
      label: import_zod.z.string(),
      name: import_zod.z.string(),
      type: import_zod.z.enum(["text", "email", "password", "number"]).nullable(),
      placeholder: import_zod.z.string().nullable(),
      value: import_zod.z.string().nullable(),
      checks: validationCheckSchema,
      validateOn: validateOnSchema
    }),
    events: ["submit", "focus", "blur"],
    description: "Text input field. Use { $bindState } on value for two-way binding. Use checks for validation (e.g. required, email, minLength). validateOn controls timing (default: blur).",
    example: {
      label: "Email",
      name: "email",
      type: "email",
      placeholder: "you@example.com"
    }
  },
  Textarea: {
    props: import_zod.z.object({
      label: import_zod.z.string(),
      name: import_zod.z.string(),
      placeholder: import_zod.z.string().nullable(),
      rows: import_zod.z.number().nullable(),
      value: import_zod.z.string().nullable(),
      checks: validationCheckSchema,
      validateOn: validateOnSchema
    }),
    description: "Multi-line text input. Use { $bindState } on value for binding. Use checks for validation. validateOn controls timing (default: blur)."
  },
  Select: {
    props: import_zod.z.object({
      label: import_zod.z.string(),
      name: import_zod.z.string(),
      options: import_zod.z.array(import_zod.z.string()),
      placeholder: import_zod.z.string().nullable(),
      value: import_zod.z.string().nullable(),
      checks: validationCheckSchema,
      validateOn: validateOnSchema
    }),
    events: ["change"],
    description: "Dropdown select input. Use { $bindState } on value for binding. Use checks for validation. validateOn controls timing (default: change)."
  },
  Checkbox: {
    props: import_zod.z.object({
      label: import_zod.z.string(),
      name: import_zod.z.string(),
      checked: import_zod.z.boolean().nullable(),
      checks: validationCheckSchema,
      validateOn: validateOnSchema
    }),
    events: ["change"],
    description: "Checkbox input. Use { $bindState } on checked for binding. Use checks for validation. validateOn controls timing (default: change)."
  },
  Radio: {
    props: import_zod.z.object({
      label: import_zod.z.string(),
      name: import_zod.z.string(),
      options: import_zod.z.array(import_zod.z.string()),
      value: import_zod.z.string().nullable(),
      checks: validationCheckSchema,
      validateOn: validateOnSchema
    }),
    events: ["change"],
    description: "Radio button group. Use { $bindState } on value for binding. Use checks for validation. validateOn controls timing (default: change)."
  },
  Switch: {
    props: import_zod.z.object({
      label: import_zod.z.string(),
      name: import_zod.z.string(),
      checked: import_zod.z.boolean().nullable(),
      checks: validationCheckSchema,
      validateOn: validateOnSchema
    }),
    events: ["change"],
    description: "Toggle switch. Use { $bindState } on checked for binding. Use checks for validation. validateOn controls timing (default: change)."
  },
  Slider: {
    props: import_zod.z.object({
      label: import_zod.z.string().nullable(),
      min: import_zod.z.number().nullable(),
      max: import_zod.z.number().nullable(),
      step: import_zod.z.number().nullable(),
      value: import_zod.z.number().nullable()
    }),
    events: ["change"],
    description: "Range slider input. Use { $bindState } on value for binding."
  },
  // ==========================================================================
  // Action Components
  // ==========================================================================
  Button: {
    props: import_zod.z.object({
      label: import_zod.z.string(),
      variant: import_zod.z.enum(["primary", "secondary", "danger"]).nullable(),
      disabled: import_zod.z.boolean().nullable()
    }),
    events: ["press"],
    description: "Clickable button. Bind on.press for handler.",
    example: { label: "Submit", variant: "primary" }
  },
  Link: {
    props: import_zod.z.object({
      label: import_zod.z.string(),
      href: import_zod.z.string()
    }),
    events: ["press"],
    description: "Anchor link. Bind on.press for click handler."
  },
  DropdownMenu: {
    props: import_zod.z.object({
      label: import_zod.z.string(),
      items: import_zod.z.array(
        import_zod.z.object({
          label: import_zod.z.string(),
          value: import_zod.z.string()
        })
      ),
      value: import_zod.z.string().nullable()
    }),
    events: ["select"],
    description: "Dropdown menu with trigger button and selectable items. Use { $bindState } on value for selected item binding."
  },
  Toggle: {
    props: import_zod.z.object({
      label: import_zod.z.string(),
      pressed: import_zod.z.boolean().nullable(),
      variant: import_zod.z.enum(["default", "outline"]).nullable()
    }),
    events: ["change"],
    description: "Toggle button. Use { $bindState } on pressed for state binding."
  },
  ToggleGroup: {
    props: import_zod.z.object({
      items: import_zod.z.array(
        import_zod.z.object({
          label: import_zod.z.string(),
          value: import_zod.z.string()
        })
      ),
      type: import_zod.z.enum(["single", "multiple"]).nullable(),
      value: import_zod.z.string().nullable()
    }),
    events: ["change"],
    description: "Group of toggle buttons. Type 'single' (default) or 'multiple'. Use { $bindState } on value."
  },
  ButtonGroup: {
    props: import_zod.z.object({
      buttons: import_zod.z.array(
        import_zod.z.object({
          label: import_zod.z.string(),
          value: import_zod.z.string()
        })
      ),
      selected: import_zod.z.string().nullable()
    }),
    events: ["change"],
    description: "Segmented button group. Use { $bindState } on selected for selected value."
  },
  Pagination: {
    props: import_zod.z.object({
      totalPages: import_zod.z.number(),
      page: import_zod.z.number().nullable()
    }),
    events: ["change"],
    description: "Page navigation. Use { $bindState } on page for current page number."
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  shadcnComponentDefinitions
});
