// @json-render/mcp/dist/app.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/app.ts
var app_exports = {};
__export(app_exports, {
  buildAppHtml: () => buildAppHtml,
  useJsonRenderApp: () => useJsonRenderApp
});
module.exports = __toCommonJS(app_exports);

// src/use-json-render-app.ts
var import_react = require("react");
var import_ext_apps = require("@modelcontextprotocol/ext-apps");
function parseSpecFromToolResult(result) {
  const textContent = result.content?.find(
    (c) => c.type === "text"
  );
  if (!textContent?.text) return null;
  try {
    const parsed = JSON.parse(textContent.text);
    if (parsed && typeof parsed === "object" && "spec" in parsed) {
      return parsed.spec;
    }
    return parsed;
  } catch {
    return null;
  }
}
function useJsonRenderApp(options = {}) {
  const { name = "json-render", version = "1.0.0" } = options;
  const [spec, setSpec] = (0, import_react.useState)(null);
  const [loading, setLoading] = (0, import_react.useState)(true);
  const [connected, setConnected] = (0, import_react.useState)(false);
  const [error, setError] = (0, import_react.useState)(null);
  const appRef = (0, import_react.useRef)(null);
  (0, import_react.useEffect)(() => {
    const app = new import_ext_apps.App({ name, version });
    appRef.current = app;
    app.ontoolresult = (result) => {
      const parsed = parseSpecFromToolResult(result);
      if (parsed) {
        setSpec(parsed);
        setLoading(false);
      }
    };
    app.connect().then(() => {
      setConnected(true);
    }).catch((err) => {
      setError(err instanceof Error ? err : new Error(String(err)));
    });
    return () => {
      app.close().catch(() => {
      });
    };
  }, []);
  const callServerTool = (0, import_react.useCallback)(
    async (toolName, args = {}) => {
      if (!appRef.current) return;
      setLoading(true);
      try {
        const result = await appRef.current.callServerTool({
          name: toolName,
          arguments: args
        });
        const parsed = parseSpecFromToolResult(result);
        if (parsed) setSpec(parsed);
      } finally {
        setLoading(false);
      }
    },
    []
  );
  return {
    spec,
    connecting: !connected && !error,
    connected,
    error,
    loading,
    app: appRef.current,
    callServerTool
  };
}

// src/build-app-html.ts
function buildAppHtml(options) {
  const { title = "json-render", css = "", js, head = "" } = options;
  return `<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>${escapeHtml(title)}</title>
  ${head}
  <style>${css}</style>
</head>
<body>
  <div id="root"></div>
  <script type="module">${js}</script>
</body>
</html>`;
}
function escapeHtml(str) {
  return str.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;");
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  buildAppHtml,
  useJsonRenderApp
});
