// @json-render/mcp/dist/index.mjs
// src/server.ts
import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
async function getExtApps() {
  const mod = await import("@modelcontextprotocol/ext-apps/server");
  return mod;
}
async function registerJsonRenderTool(server, options) {
  const { catalog, name, title, description, resourceUri } = options;
  const { registerAppTool } = await getExtApps();
  const specZodSchema = catalog.zodSchema();
  registerAppTool(
    server,
    name,
    {
      title,
      description,
      inputSchema: { spec: specZodSchema },
      _meta: { ui: { resourceUri } }
    },
    async (args) => {
      const spec = args.spec;
      const validation = catalog.validate(spec);
      const validSpec = validation.success ? validation.data : spec;
      return {
        content: [
          {
            type: "text",
            text: JSON.stringify(validSpec)
          }
        ]
      };
    }
  );
}
async function registerJsonRenderResource(server, options) {
  const { resourceUri, html } = options;
  const { registerAppResource, RESOURCE_MIME_TYPE: mimeType } = await getExtApps();
  registerAppResource(
    server,
    resourceUri,
    resourceUri,
    { mimeType },
    async () => ({
      contents: [
        {
          uri: resourceUri,
          mimeType,
          text: html,
          _meta: {
            ui: {
              csp: {
                resourceDomains: ["https:"],
                connectDomains: ["https:"]
              }
            }
          }
        }
      ]
    })
  );
}
async function createMcpApp(options) {
  const { name, version, catalog, html, tool } = options;
  const toolName = tool?.name ?? "render-ui";
  const toolTitle = tool?.title ?? "Render UI";
  const resourceUri = `ui://${toolName}/view.html`;
  const catalogPrompt = catalog.prompt();
  const toolDescription = tool?.description ?? `Render an interactive UI. The spec argument must be a json-render spec conforming to the catalog.

${catalogPrompt}`;
  const server = new McpServer({ name, version });
  await registerJsonRenderTool(server, {
    catalog,
    name: toolName,
    title: toolTitle,
    description: toolDescription,
    resourceUri
  });
  await registerJsonRenderResource(server, { resourceUri, html });
  return server;
}
export {
  createMcpApp,
  registerJsonRenderResource,
  registerJsonRenderTool
};
