// @json-render/mcp/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  createMcpApp: () => createMcpApp,
  registerJsonRenderResource: () => registerJsonRenderResource,
  registerJsonRenderTool: () => registerJsonRenderTool
});
module.exports = __toCommonJS(index_exports);

// src/server.ts
var import_mcp = require("@modelcontextprotocol/sdk/server/mcp.js");
async function getExtApps() {
  const mod = await import("@modelcontextprotocol/ext-apps/server");
  return mod;
}
async function registerJsonRenderTool(server, options) {
  const { catalog, name, title, description, resourceUri } = options;
  const { registerAppTool } = await getExtApps();
  const specZodSchema = catalog.zodSchema();
  registerAppTool(
    server,
    name,
    {
      title,
      description,
      inputSchema: { spec: specZodSchema },
      _meta: { ui: { resourceUri } }
    },
    async (args) => {
      const spec = args.spec;
      const validation = catalog.validate(spec);
      const validSpec = validation.success ? validation.data : spec;
      return {
        content: [
          {
            type: "text",
            text: JSON.stringify(validSpec)
          }
        ]
      };
    }
  );
}
async function registerJsonRenderResource(server, options) {
  const { resourceUri, html } = options;
  const { registerAppResource, RESOURCE_MIME_TYPE: mimeType } = await getExtApps();
  registerAppResource(
    server,
    resourceUri,
    resourceUri,
    { mimeType },
    async () => ({
      contents: [
        {
          uri: resourceUri,
          mimeType,
          text: html,
          _meta: {
            ui: {
              csp: {
                resourceDomains: ["https:"],
                connectDomains: ["https:"]
              }
            }
          }
        }
      ]
    })
  );
}
async function createMcpApp(options) {
  const { name, version, catalog, html, tool } = options;
  const toolName = tool?.name ?? "render-ui";
  const toolTitle = tool?.title ?? "Render UI";
  const resourceUri = `ui://${toolName}/view.html`;
  const catalogPrompt = catalog.prompt();
  const toolDescription = tool?.description ?? `Render an interactive UI. The spec argument must be a json-render spec conforming to the catalog.

${catalogPrompt}`;
  const server = new import_mcp.McpServer({ name, version });
  await registerJsonRenderTool(server, {
    catalog,
    name: toolName,
    title: toolTitle,
    description: toolDescription,
    resourceUri
  });
  await registerJsonRenderResource(server, { resourceUri, html });
  return server;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createMcpApp,
  registerJsonRenderResource,
  registerJsonRenderTool
});
